CONFIGURATION.md

```diff
@@ -0,0 +1,293 @@
+2. if the strain configuration can be modified by the deployment action, a new strain can be introduced, but as the `helix-config.yaml` is the single point of synchronization, deployment in one branch will automatically deactivate all strains that refer to concurrent deployments made in different branches. In active development, this will lead to race conditions and intermittent test failures because the strain you just deployed got "undeployed" by a build triggered in another branch.
+3. even within one branch, making two committs in short succession will lead to the second commit tainting the deployment of the first build, with potentially misleading results. For instance a bug fix might get attributed to the first commit because this is where testing first succeeds, although the fix was deployed in the second commit
+
+
+### Proposal: Temporary Strains
+
+For testing a deployment in a Continuous Integration environment, it can be useful to have strains that are not persisted in the `helix-config.yaml`, but can still be activated for testing.
+
+To enable temporary strains, the strain resolution logic in VCL will be modified so that when an `X-Strain` cookie or header is present and the value of the header contains a `/`, both the `X-Strain` (name) and `X-Tag` (tag) will be parsed. 
+For the most part, the default logic of the current `X-Strain` will be applied, so that the strain's directory index, static repo, etc. will be used. The only exception is the resolution of the OpenWhisk action to execute.
+Here, the `-git--([\w]+)--` pattern will be replaced with the `X-Tag` value, effectively pinning the used action to the tagged deployment. 
+
+As this will lead to a large number of temporary actions in OpenWhisk, we also introduce a `hlx undeploy` command that clears a temporary deployment made earlier. This can be run at the end of every CI job.
+
+`hlx test` and `hlx perf` should use the new temporary strains when running in CI.
+
+## Discussion: layout
+
+During the 4/2018 hackathon we also discussed the layout of the project. it was decided that:
+
+- templates and pre functions will be in `./src`
+- client side javascript **must not** go into `./src`. 
+- we will provide a react example, and put the sources in `./react`
+- all static files will go to `htdocs`
+- all additional openwhisk actions will go to `htdocs/cgi-bin`
+
+## Discussion: action names
+
+The deployment mechanism changes so that the **sha of the code repository** at deployment time will become the package name. the action name will simply be the script name without any extensions.
+ 
+for example: 
+
+legacy name:
+```
+/helix/https---github-com-trieloff-helix-demo-git--master--html
+```
+
+new name:
+
+```
+/helix/dcfee99d45c6a2baed86c8e4921623c48b902522/html
+```
+
```

Comment:
Question: for the managed case, we wanted to put all actions to the same namespace `helix-clients` and group them with packages by customer name. openwhisk can't have sub-folders in packages. so I would rather not use up the package for the SHA here, so that the packages still can be used for clients.

```suggest
/helix/dcfee99d45c6a2baed86c8e4921623c48b902522--html
```